src/Servers/IIS/IIS/src/Core/IISHttpContext.FeatureCollection.cs

```diff
@@ -86,30 +86,14 @@ private void ExtraFeatureSet(Type key, object value)
 
         string IHttpRequestFeature.Protocol
         {
-            get
+            get => _httpProtocolVersion ??= HttpVersion switch
             {
-                if (_httpProtocolVersion == null)
-                {
-                    var protocol = HttpVersion;
-                    if (protocol.Major == 1 && protocol.Minor == 1)
-                    {
-                        _httpProtocolVersion = "HTTP/1.1";
-                    }
-                    else if (protocol.Major == 1 && protocol.Minor == 0)
-                    {
-                        _httpProtocolVersion = "HTTP/1.0";
-                    }
-                    else
-                    {
-                        _httpProtocolVersion = "HTTP/" + protocol.ToString(2);
-                    }
-                }
-                return _httpProtocolVersion;
-            }
-            set
-            {
-                _httpProtocolVersion = value;
-            }
+                { Major: 2, Minor: 0 } => "HTTP/2",
```

Comment:
I thought about using the same approach as in https://github.com/aspnet/AspNetCore/blob/c74e9efd0e6ea5b4d6737d23b92042772a50a249/src/Servers/HttpSys/src/FeatureContext.cs#L177, but

* this code is much more readable (the major/minor-matrix can be read easily)
* it's an explicit interface property that is lazy initialized, so the perf-difference between and object-comparison and two int comparisons should not show up
* in the case that `HttpVersion` is set to a different `Version`-instance (i.e. set not via `Constants.V2`) `HTTP/2.0` may be returned, so here it is the safer way, as `HTTP/2` will be returned

So I'm tempted to update https://github.com/aspnet/AspNetCore/blob/c74e9efd0e6ea5b4d6737d23b92042772a50a249/src/Servers/HttpSys/src/FeatureContext.cs#L174-L193 to the pattern used here. Should I?

Reply:
Yes, feel free to update HttpSys. Or better yet, move this to an API in the code that HttpSys and IIS share.

Reply:
> API in the code that HttpSys and IIS share

I can't find one that is shared 😢 

An extension method 
```c#
public static string GetHttpProtocolVersion(this Version version);
```
would do the job, but where to put this one?

Reply:
This code is all shared: https://github.com/aspnet/AspNetCore/tree/master/src/Shared/HttpSys